sr_robot_launch/scripts/construct_robot_description

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python
+
+# Copyright 2019 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import rospy
+import rospkg
+from subprocess import check_output
+from sr_ur_arm_calibration_loader.sr_ur_arm_calibration_loader import SrUrLoadCalibration
+
+
+class SrConstructRobotDescription():
+    def __init__(self):
+        self._robot_description_file = rospy.get_param("~robot_description_file")
+        CONST_ARM_TYPE = rospy.get_param("~arm_type")
+        self._side = ''
+        self._robot_description_params = {}
+        self._node_name = rospy.get_name()
+        self._arm_ra = {'prefix': 'ra', 'ip_address': '192.168.1.1', 'arm_type': CONST_ARM_TYPE}
+        self._arm_la = {'prefix': 'la', 'ip_address': '192.168.2.1', 'arm_type': CONST_ARM_TYPE}
+
+        if 'bimanual' in self._robot_description_file:
+            self._side = 'both'
+        else:
+            prefix = rospy.get_param("~prefix")
+            if 'rh' in prefix:
+                self._side = 'right'
+            elif 'lh' in prefix:
+                self._side = 'left'
+            else:
+                rospy.logerr("'prefix' param not specified correctly")
+
+        self._process_parameters()
+        urdf = self._parse_xacro()
+        rospy.set_param('robot_description', urdf)
+
+    def _parse_xacro(self):
+        xacro_commands = []
+        xacro_commands.append("xacro")
+        xacro_commands.append(self._robot_description_file)
+        for key, value in self._robot_description_params.items():
+            xacro_commands.append(key + ':=' + str(value))
+        return self._generate_urdf(xacro_commands)
+
+    def _process_parameters(self):
```

Comment:
this is being very anal about the topic, but would look nice if you define methods in order of calling them so its the easiest to read the code :)